src/store/web3wallet.store.ts

```diff
@@ -12,8 +12,24 @@ export const coreAtom = atom(() => {
   });
 });
 
+export const relayerConnectionStatusAtom = atom("disconnected"); // Create a new atom to store the connection status
+
 export const web3walletAtom = atom((get) => {
   const core = get(coreAtom);
+  const [_connectionStatus, setConnectionStatus] = useAtom(
+    relayerConnectionStatusAtom,
+  );
+
+  core.relayer.on("relayer_connect", () => {
+    setConnectionStatus("connected");
+  });
+
+  core.relayer.on("relayer_disconnect", () => {
+    setConnectionStatus("disconnected");
+    core.relayer.restartTransport().catch(() => {
+      console.error("couldn't restart transport")
+    });
+  });
```

Comment:
This should probably leave in a react hook instead of inside the atom itself, as you are not supposed to use an hook in the atom and it might be better to do the effects of listening to events in a `useEffect`

Reply:
`onMount` could be an option too probably https://jotai.org/docs/core/atom#onmount-property

You could also use another derivedAtom to just set the connection status when relayer changes, but in the end it's kind of the same as using a react hook and I don't know if there is some downsides
```ts
export const connectionStatusAtom = atom("disconnected"); // Create a new atom to store the connection status

export const relayerConnectionStatusAtom = atom(
  (get) => get(connectionStatusAtom),
  (get, set) => {
    const core = get(coreAtom);

    core.relayer.on("relayer_connect", () => {
      set(connectionStatusAtom, "connected");
    });

    core.relayer.on("relayer_disconnect", () => {
      set(connectionStatusAtom, "disconnected");
      core.relayer.restartTransport().catch(() => {
        console.error("couldn't restart transport"); // TODO: Maybe use a toast instead
      });
    });
  },
);
```

Reply:
Done here: https://github.com/LedgerHQ/wallet-connect-live-app/pull/168/commits/557788d9e9e6b92ab6c19f059464c8155a453aec

Reply:
Yes because you'd want to get the core before getting the atom for web3wallet so you can listen before you init the web3wallet

Also you could init the connectionStatus with the core relayer state